src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -145,7 +145,16 @@ public IEnumerable<Assembly> Assemblies
             {
                 foreach (Assembly a in GetLoadedAssemblies())
                 {
-                    AssemblyLoadContext alc = GetLoadContext(a);
+                    AssemblyLoadContext alc;
+                    try
+                    {
+                        alc = GetLoadContext(a);
+                    }
+                    catch
+                    {
```

Comment:
I do not think that this will work. You cannot catch bad AVs and crashes in managed code. We consider them fatal.